### PR TITLE
Add UI Event linking to DraggableColorbar

### PR DESCRIPTION
**Pull request by ruuskas**
This change makes the DraggableColorbar class work with UI Events. Users can now link figures that contain topoplots or epoch/tfr images and manipulate the color scale and color map interactively.

Basic example based on the [Using the event system to link figures](https://mne.tools/dev/auto_tutorials/visualization/20_ui_events.html#sphx-glr-auto-tutorials-visualization-20-ui-events-py) tutorial.

```
import mne
from mne.viz.ui_events import link

# Turn on interactivity
plt.ion()
data_path = mne.datasets.sample.data_path()
evokeds_fname = data_path / "MEG" / "sample" / "sample_audvis-ave.fif"
evokeds = mne.read_evokeds(evokeds_fname)

left_auditory = evokeds[0]
right_auditory = evokeds[1]

fig1 = left_auditory.plot_topomap("interactive")
fig2 = right_auditory.plot_topomap("interactive")

link(fig1, fig2)
```

### CODE REVIEW

**Comment by larsoner** [comment]
... also needs a `doc/changes/devel.rst` update to mention this enhancement

File: mne/viz/ui_events.py [hunk 1]
```diff
@@ -122,18 +124,23 @@ class ColormapRange(UIEvent):
     kind : str
         Kind of colormap being updated. The Notes section of the drawing
         routine publishing this event should mention the possible kinds.
+    sensor_type : str
```
Comment (wmvanvliet) [inline mne/viz/ui_events.py:127]:
```suggestion
    ch_type : str
```

Comment (wmvanvliet) [reply mne/viz/ui_events.py:127]: `ch_type` and `cmap` also need to be added under the `Parameters` section of this docstring.

**Comment by wmvanvliet** [comment]
CI failure "Test Windows 3.11 pip pre" is not releated to this PR

**Comment by drammock** [comment]
> CI failure "Test Windows 3.11 pip pre" is not releated to this PR

agreed, @larsoner snuck a fix into #12064

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -958,7 +958,7 @@ def _plot_image(
         cbar = plt.colorbar(im, ax=ax)
         cbar.ax.set_title(ch_unit)
         if cmap[1]:
-            ax.CB = DraggableColorbar(cbar, im)
+            ax.CB = DraggableColorbar(cbar, im, "image", this_type)
```
Comment (wmvanvliet) [inline mne/viz/evoked.py:961]:
```suggestion
            ax.CB = DraggableColorbar(cbar, im, "evoked_image", this_type)
```

File: mne/viz/topo.py [hunk 3]
```diff
@@ -457,7 +457,7 @@ def _imshow_tfr(
         else:
             cbar = plt.colorbar(mappable=img, ax=ax)
         if interactive_cmap:
-            ax.CB = DraggableColorbar(cbar, img)
+            ax.CB = DraggableColorbar(cbar, img, kind=None, ch_type=None)
```
Comment (wmvanvliet) [inline mne/viz/topo.py:460]:
```suggestion
            ax.CB = DraggableColorbar(cbar, img, kind="tfr_image", ch_type=None)
```
can we give a `kind` here? something like this?

**Review by wmvanvliet — COMMENTED** [review]
Looking great. Only a few nickpicks regarding `kind`

**Comment by ruuskas** [comment]
@wmvanvliet Any idea why the CI doctest fails?

**Comment by wmvanvliet** [comment]
seems unrelated. Let's try again with latest `main` merged in.

**Comment by ruuskas** [comment]
This should be good now @wmvanvliet

**Pull request merged**